common/oidc-client/oauth2-proxy/README.md

```diff
@@ -0,0 +1,213 @@
+   with a JWT bearer token. `oauth2-proxy` will automatically trust the JWT
+   issued by the configured oidc provider, which in the default setup is dex.
+   Additional trusted JWTs can be configured in a way that `oauth2-proxy` will
+   skip the authentication and forward the request back to istio including
+   provided authorization header.
+
+   The ultimate goal of this step is to provide the `Authorization` header
+   with JWT Bearer Token which is later used in `RequestAuthentication` to
+   configure Istio to trust this JWT, parse JWT Claims to include user email
+   and groups in custom Kubeflow authorization headers, allow routing and
+   authorization decision based on JWT Claims.
+4. In this step Istio will always have the JWT in the request making it possible
+   to make authorization decision based on JWT Claims.
+
+## Using HTTPS
+`oauth2-proxy` is configured with `http` endpoint secured by Istio Service Mesh.
+Because of that, `oauth2-proxy` will guess that the auth redirect uri is supposed
+to be handler with `http` as well. To force using `https`, change variable
+`FORCE_HTTPS` to `true` in `kustomization.yaml`. This will use `oauth2-proxy`
+configuration option `--cookie-secure` which will redirect with `https`.
+
+## Istio JWT Pub Key Refresh Interval
+On initial Kubeflow setup, it's highly possible that `istiod` will become
+available before `dex`. Using Istio `RequestAuthentication` configures Istio to call
+the Issuer URL to gather the JWT Pub Key. If Issuer is not yet available, placeholder
+keys will be configured and the setup will become unusable until Istio has access
+to the correct JWT Pub Key. For that reason, `istiod` is configured with env variable
+`PILOT_JWT_PUB_KEY_REFRESH_INTERVAL="1m"` so the JWT Pub Key is refreshed every
+minute instead of the default 20 minutes.
+
+If this is not configured, user will be presented with following Istio error:
+```
+Jwks doesn't have key to match kid or alg from Jwt
+```
+
+## Issues with this setup
+While not directly an Istio or oauth2-proxy issue, currently Kubeflow is configured
+to always redirect to the URL provided in logout response body object under key
+`afterLogoutURL`. This is custom integration with `oidc-authservice` component.
+oauth2-proxy has redirect capabilities and can redirect to the base Kubeflow
+page but because of this custom integration, currently logging out will remove
+the authentication cookie but not redirect to the Kubeflow Home Page.
+
+The details of this custom integration can be found here:
+* https://github.com/arrikto/oidc-authservice/blob/0c4ea9a/server.go#L509
+* https://github.com/kubeflow/kubeflow/blob/c6c4492/components/centraldashboard/public/components/logout-button.js#L50
+
+To login again, user have to refresh the page.
+
+## Example installation
+To install Kubeflow configured to use `oauth2-proxy` with `istio`
+`envoyExtAuthzHttp` extension, following changes has to be done to the
```

Comment:
You are very often missing "the" in your sentences within the readme. For example "extension, following changes has to be done to the" should be "extension, THE following changes HAVE to be applied to the"